packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,294 @@
+import { observable } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    FoundationElement
+} from '@microsoft/fast-foundation';
+import {
+    keyTab,
+    keyEnter,
+    keyArrowDown,
+    keyArrowUp,
+    keyEscape
+} from '@microsoft/fast-web-utilities';
+import type { SuggestionProps } from '@tiptap/suggestion';
+import type { ListOption } from '../../list-option';
+import type { Listbox } from '../../listbox';
+import type { MentionDetail } from '../editor/types';
+import { styles } from './styles';
+import { template } from './template';
+import type { AnchoredRegion } from '../../anchored-region';
+import type { MentionExtensionConfiguration } from '../models/mention-extension-configuration';
+import type { MappingConfigs } from '../../rich-text-mention/base/types';
+import { normalizeString } from '../../utilities/models/string-normalizer';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-rich-text-mention-list-box': RichTextMentionListBox;
+    }
+}
+
+/**
+ * A rich text mention list box which acts as a popup for "@mention" support in editor
+ */
+export class RichTextMentionListBox extends FoundationElement {
+    /**
+     * @internal
+     */
+    @observable
+    public region?: AnchoredRegion;
+
+    /**
+     * @internal
+     */
+    @observable
+    public activeCharacter?: string;
+
+    /**
+     * @internal
+     */
+    @observable
+    public activeMappingConfigs?: MappingConfigs;
+
+    /**
+     * @internal
+     */
+    @observable
+    public open?: boolean;
+
+    /**
+     * @internal
+     */
+    @observable
+    public readonly childItems: Element[] = [];
+
+    /**
+     * @internal
+     */
+    @observable
+    public filter?: string;
+
+    /**
+     * @internal
+     */
+    public listBox!: Listbox;
+
+    @observable
+    private anchorElement?: HTMLElement;
+
+    private hasAnySelectableOption = false;
+
+    private mentionExtensionConfig?: MentionExtensionConfiguration[];
+
+    private suggestionProps!: SuggestionProps;
+
+    private readonly intersectionObserver: IntersectionObserver = new IntersectionObserver(
```

Comment:
We have introduced intersection observer here to close the mention popup, when the anchor element is hidden.

This is based on the issue that we faced with the `menu-button` in the comments functionality. The issue is, when we open the Menu and scroll through the comments list, the opened `menu button` appears on top of the entire page.
![image](https://github.com/ni/nimble/assets/123383804/83eed817-b765-4f19-b139-52caedfc02dc)

More details: https://teams.microsoft.com/l/message/19:0c59ac4a4cc1438894cb42f4b21703f9@thread.skype/1692631178768?tenantId=87ba1f9a-44cd-43a6-b008-6fdb45a5204e&groupId=41626d4a-3f1f-49e2-abdc-f590be4a329d&parentMessageId=1692631178768&teamName=ASW%20SystemLink&channelName=UI%20Working%20Group&createdTime=1692631178768

We suspect the same issue with the `mention-list` as well. Hence we handled it using the `intersection observer`.

Please let me know your thoughts.

Reply:
If that's the pattern we've used before, it makes sense to me.